tests/core/pyspec/eth2spec/test/altair/light_client/test_data_collection.py

```diff
@@ -0,0 +1,1132 @@
+    if attested_period == signature_period:
+        assert update_attested_bid(get_light_client_update_for_period(test, attested_period).data) == bids_a[-1]
+    else:
+        assert signature_period == attested_period + 1
+        assert update_attested_bid(get_light_client_update_for_period(test, attested_period).data) == bids_a[-2]
+        assert get_light_client_update_for_period(test, signature_period).spec is None
+    assert update_attested_bid(get_light_client_finality_update(test).data) == bids_a[-1]
+    assert update_attested_bid(get_light_client_optimistic_update(test).data) == bids_a[-1]
+    bids_a.append(bid_a)
+
+    # Data has been restored
+    state_a_period = spec_a.compute_sync_committee_period_at_slot(state_a.slot)
+    for period in range(state_period + 1, state_a_period):
+        assert get_light_client_update_for_period(test, period).spec is not None
+
+    # Finish test
+    yield from finish_test(test)
+
+
+@with_phases(phases=[BELLATRIX], other_phases=[CAPELLA, DENEB])
+@spec_test
+@with_config_overrides({
+    'CAPELLA_FORK_EPOCH': 1 * 8,  # SyncCommitteePeriod 1
+    'DENEB_FORK_EPOCH': 2 * 8,  # SyncCommitteePeriod 2
+}, emit=False)
+@with_state
+@with_matching_spec_config(emitted_fork=DENEB)
+@with_presets([MINIMAL], reason="too slow")
+def test_capella_deneb_reorg_aligned(spec, phases, state):
+    yield from run_test_multi_fork(spec, phases, state, CAPELLA, DENEB)
+
+
+@with_phases(phases=[BELLATRIX], other_phases=[CAPELLA, DENEB])
+@spec_test
+@with_config_overrides({
+    'CAPELLA_FORK_EPOCH': 1 * 8 + 4,  # SyncCommitteePeriod 1 (+ 4 epochs)
+    'DENEB_FORK_EPOCH': 3 * 8 + 4,  # SyncCommitteePeriod 3 (+ 4 epochs)
+}, emit=False)
+@with_state
+@with_matching_spec_config(emitted_fork=DENEB)
+@with_presets([MINIMAL], reason="too slow")
+def test_capella_deneb_reorg_unaligned(spec, phases, state):
+    yield from run_test_multi_fork(spec, phases, state, CAPELLA, DENEB)
+
+
+@with_phases(phases=[CAPELLA], other_phases=[DENEB, ELECTRA])
+@spec_test
+@with_config_overrides({
+    'DENEB_FORK_EPOCH': 1 * 8,  # SyncCommitteePeriod 1
+    'ELECTRA_FORK_EPOCH': 2 * 8,  # SyncCommitteePeriod 2
+}, emit=False)
+@with_state
+@with_matching_spec_config(emitted_fork=ELECTRA)
+@with_presets([MINIMAL], reason="too slow")
+def test_deneb_electra_reorg_aligned(spec, phases, state):
+    yield from run_test_multi_fork(spec, phases, state, DENEB, ELECTRA)
+
+
+@with_phases(phases=[CAPELLA], other_phases=[DENEB, ELECTRA])
+@spec_test
+@with_config_overrides({
+    'DENEB_FORK_EPOCH': 1 * 8 + 4,  # SyncCommitteePeriod 1 (+ 4 epochs)
+    'ELECTRA_FORK_EPOCH': 3 * 8 + 4,  # SyncCommitteePeriod 3 (+ 4 epochs)
+}, emit=False)
+@with_state
+@with_matching_spec_config(emitted_fork=ELECTRA)
+@with_presets([MINIMAL], reason="too slow")
+def test_deneb_electra_reorg_unaligned(spec, phases, state):
+    yield from run_test_multi_fork(spec, phases, state, DENEB, ELECTRA)
```

Comment:
This test should be moved to `capella/light_client/` folder, as well as test `test_sync.py` ones.

Reply:
Let's handle this in a separate PR to be included in the next release. I started to make these changes but realized it wasn't as straightforward as I thought.

Reply:
Have applied the changes to both data collection and test_sync as requested.